packages/stack/blockchain-client/src/index.js

```diff
@@ -19,6 +20,16 @@ class BlockchainClient {
         if (err) {
           return cb(err);
         }
+        if (endpoint.startsWith('ws')) {
+          return cb(null, new Web3.providers.WebsocketProvider(endpoint, {
+            headers: {Origin: constants.embarkResourceOrigin},
+            // TODO remove this when Geth fixes this: https://github.com/ethereum/go-ethereum/issues/16846
```

Comment:
Most surely an admirable effort in identifying the problem and providing a fix, awesome bug-hunting skills! 🐛 💀 

I think the question is raised, though, what policy we want to / should have re: geth versions Embark supports? Ideally, that's somehow related to the "real world" setups of our long-time and new users.